hoodie-common/src/main/java/com/uber/hoodie/common/util/HoodieAvroUtils.java

```diff
@@ -102,7 +102,7 @@ public static Schema addMetadataFields(Schema schema) {
     }
 
     Schema mergedSchema = Schema
-        .createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), false);
+        .createRecord(schema.getName()+"_table", schema.getDoc(), schema.getNamespace(), false);
```

Comment:
Can you explain the need to add this string to the schema ?

Reply:
when you have the SpecificRecord implementation on the classpath the avro parquet implementation will resolve the implementation class as the schema for the table and the metadata fields doen't exist in the generated classes so it result as type exception.  My test prove the bug.  

that's the exception without the fix:
Caused by: java.lang.ClassCastException: org.apache.avro.util.Utf8 cannot be cast to java.lang.String
	at com.huber.hoodie.example.ExampleValue.put(ExampleValue.java:43)
	at org.apache.avro.generic.GenericData.setField(GenericData.java:651)
	at org.apache.parquet.avro.AvroRecordConverter.set(AvroRecordConverter.java:295)
	at org.apache.parquet.avro.AvroRecordConverter$1.add(AvroRecordConverter.java:109)
	at org.apache.parquet.avro.AvroConverters$BinaryConverter.addValueFromDictionary(AvroConverters.java:81)
	at org.apache.parquet.column.impl.ColumnReaderImpl$1.writeValue(ColumnReaderImpl.java:175)
	at org.apache.parquet.column.impl.ColumnReaderImpl.writeCurrentValueToConverter(ColumnReaderImpl.java:371)
	at org.apache.parquet.io.RecordReaderImplementation.read(RecordReaderImplementation.java:405)
	at org.apache.parquet.hadoop.InternalParquetRecordReader.nextKeyValue(InternalParquetRecordReader.java:218)
	... 34 more

Reply:
Okay makes sense.. we mostly use GenericRecord here.  @n3nash will take this home and I can merge it in.

Reply:
I see, in that case should we just change the namespace ? Just put an empty namespace. WDYT ?

Reply:
but if we do that we could have a name conflict in the sparkcontext as hoodie register the avro schema in it

Reply:
why ? hoodie registers one avro schema in the sparkcontext, if there are different schemas they have different schema names and so an empty namespace should still be okay. And this will not conflict with SpecificRecords since they will fall back to namespace resolution anyways. Thoughts ?

Reply:
But if I use hoodie in my own job spark I couldn't register my own avro schema because I use it in hoodie?

Reply:
I think @n3nash is just asking if empty namespace will work, instead of adding the suffix.. 
I am not super familiar with backwards compatibility aspects of this as well. guess it should be fine? (i.e does it play well with older parquet data written with current namespace?)

I have one suggestion .. can we name the suffix `_hoodie` instead of `_table`

Reply:
@vinothchandar Yeah, not sure about backwards compatibility either. I think `_table` is fine by me.

Reply:
you're not going to have issue with the compatibility as the avro schema is external

Reply:
please change to "_hoodie" too.